src/Order/Order.php

```diff
@@ -0,0 +1,547 @@
+<?php
+declare(strict_types=1);
+
+namespace SwipeStripe\Order;
+
+use Money\Money;
+use SilverStripe\Control\Director;
+use SilverStripe\Core\CoreKernel;
+use SilverStripe\Forms\FieldGroup;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Omnipay\Model\Payment;
+use SilverStripe\Omnipay\Service\ServiceResponse;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\DB;
+use SilverStripe\ORM\FieldType\DBBoolean;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\ORM\FieldType\DBEnum;
+use SilverStripe\ORM\FieldType\DBVarchar;
+use SilverStripe\ORM\HasManyList;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\Permission;
+use SwipeStripe\CMSHelper;
+use SwipeStripe\Order\Cart\ViewCartPage;
+use SwipeStripe\Order\OrderItem\OrderItem;
+use SwipeStripe\Order\OrderItem\OrderItemAddOn;
+use SwipeStripe\ORM\FieldType\DBAddress;
+use SwipeStripe\Price\DBPrice;
+use SwipeStripe\Price\PriceField;
+use SwipeStripe\Price\SupportedCurrencies\SupportedCurrenciesInterface;
+use SwipeStripe\ShopPermissions;
+
+/**
+ * Class Order
+ * @package SwipeStripe\Order
+ * @property bool $IsCart
+ * @property bool $CartLocked
+ * @property string $GuestToken
+ * @property string $Hash
+ * @property string $CustomerName
+ * @property string $CustomerEmail
+ * @property DBAddress $BillingAddress
+ * @property string $ConfirmationTime
+ * @property string $Environment
+ * @method HasManyList|OrderItem[] OrderItems()
+ * @method HasManyList|OrderAddOn[] OrderAddOns()
+ * @mixin Payable
+ */
+class Order extends DataObject
+{
+    const GUEST_TOKEN_BYTES = 16;
+
+    /**
+     * @var string
+     */
+    private static $table_name = 'SwipeStripe_Order';
+
+    /**
+     * @var array
+     */
+    private static $db = [
+        'IsCart'           => DBBoolean::class,
+        'CartLocked'       => DBBoolean::class,
+        'GuestToken'       => DBVarchar::class,
+        'ConfirmationTime' => DBDatetime::class,
+        'Environment'      => DBEnum::class . '(array("' . CoreKernel::DEV . '", "' . CoreKernel::TEST . '", "' . CoreKernel::LIVE . '"), "' . CoreKernel::DEV . '")',
```

Comment:
Very unsure about this. This has a strong code smell. A site + db should only ever be on one environment.